src/lib/concurrency/transaction_manager.cpp

```diff
@@ -23,7 +23,37 @@ TransactionManager::TransactionManager()
 CommitID TransactionManager::last_commit_id() const { return _last_commit_id; }
 
 std::shared_ptr<TransactionContext> TransactionManager::new_transaction_context() {
-  return std::make_shared<TransactionContext>(_next_transaction_id++, _last_commit_id);
+  std::unique_lock<std::mutex> lock(_mutex_active_snapshot_commit_ids);
```

Comment:
Intel writes following [in the docs](https://software.intel.com/en-us/node/506181):
> concurrent_unordered_set and concurrent_unordered_multiset support concurrent insertion and traversal, but not concurrent erasure.

Since we are also erasing elements from the multiset, we cannot get rid of the locks, it seems.

Reply:
Got it.